File: wagerproof-ios-native/Wagerproof/Features/Agents/Sheets/AgentHRBottomSheet.swift
import SwiftUI
import WagerproofDesign
import WagerproofModels

/// Native port of `components/agents/AgentHRBottomSheet.tsx`.
///
/// "HR DEPARTMENT" comparison sheet: ranks every agent that has settled picks
/// by net units, grades them, and surfaces a recommendation. Includes a
/// "fire to save $$$" panel for losers. Pure visualization — no mutations.
///
/// Presented from `CompanyDashboardBanner` (or wherever the host wants to
/// drop the HR CTA). Uses SwiftUI's `.sheet` with `.medium` / `.large`
/// detents — native idiomatic replacement for Gorhom bottom-sheet.
struct AgentHRBottomSheet: View {
    let agents: [AgentWithPerformance]
    var onDismiss: () -> Void = {}

    private static let unitSize: Double = 100

    enum Grade: String {
        case s = "S", a = "A", b = "B", c = "C", d = "D", f = "F"

        var color: Color {
            switch self {
            case .s: return Color(hex: 0xFFD700)
            case .a: return Color(hex: 0x00E676)
            case .b: return Color(hex: 0x69F0AE)
            case .c: return Color(hex: 0xFFC107)
            case .d: return Color(hex: 0xFF9800)
            case .f: return Color(hex: 0xFF5252)
            }
        }
    }

    private struct ReportCard: Identifiable {
        let agent: AgentWithPerformance
        let grade: Grade
        let netUnits: Double
        let dollarImpact: Double
        let winRate: Double?
        let record: String
        let companyWithout: Double
        let recommendation: String
        let isCostingMoney: Bool

        var id: String { agent.id }
    }

    private var cards: [ReportCard] {
        let totalNetUnits = agents.reduce(0) { $0 + ($1.performance?.netUnits ?? 0) }
        let totalBankroll = 1000 + totalNetUnits * Self.unitSize

        return agents
            .filter { ($0.performance?.totalPicks ?? 0) > 0 }
            .map { agent in
                let perf = agent.performance!
                let settled = perf.wins + perf.losses
                let winRate = settled > 0 ? Double(perf.wins) / Double(settled) * 100 : nil
                let grade = Self.grade(forNetUnits: perf.netUnits)
                let companyWithout = totalBankroll - perf.netUnits * Self.unitSize
                var card = ReportCard(
                    agent: agent,
                    grade: grade,
                    netUnits: perf.netUnits,
                    dollarImpact: perf.netUnits * Self.unitSize,
                    winRate: winRate,
                    record: perf.recordLabel,
                    companyWithout: companyWithout,
                    recommendation: "",
                    isCostingMoney: perf.netUnits < 0
                )
                card = ReportCard(
                    agent: agent,
                    grade: grade,
                    netUnits: perf.netUnits,
                    dollarImpact: perf.netUnits * Self.unitSize,
                    winRate: winRate,
                    record: perf.recordLabel,
                    companyWithout: companyWithout,
                    recommendation: Self.recommendation(
                        grade: grade,
                        name: agent.agent.name,
                        netUnits: perf.netUnits,
                        winRate: winRate
                    ),
                    isCostingMoney: perf.netUnits < 0
                )
                return card
            }
            .sorted { $0.netUnits < $1.netUnits }
    }

    private var winners: [ReportCard] { cards.filter { !$0.isCostingMoney }.sorted { $0.netUnits > $1.netUnits } }
    private var losers: [ReportCard] { cards.filter { $0.isCostingMoney } }
    private var winnersTotal: Double { winners.reduce(0) { $0 + $1.dollarImpact } }
    private var losersTotal: Double { losers.reduce(0) { $0 + $1.dollarImpact } }
    private var totalBankroll: Double {
        let totalNetUnits = agents.reduce(0) { $0 + ($1.performance?.netUnits ?? 0) }
        return 1000 + totalNetUnits * Self.unitSize
    }

    var body: some View {
        NavigationStack {
            ScrollView {
                VStack(alignment: .leading, spacing: 16) {
                    headerBlock

                    if winners.isEmpty && losers.isEmpty {
                        emptyState
                    }

                    if !winners.isEmpty {
                        summaryCard(
                            title: "WINNERS",
                            tint: Color(hex: 0x00E676),
                            total: winnersTotal,
                            body: "\(winners.count) agent\(winners.count == 1 ? "" : "s") earning money for the company."
                        )
                        ForEach(winners) { card in
                            reportRow(card)
                        }
                    }

                    if !losers.isEmpty {
                        summaryCard(
                            title: "LOSERS",
                            tint: Color(hex: 0xFF5252),
                            total: losersTotal,
                            body: losersSummary
                        )
                        ForEach(losers) { card in
                            reportRow(card)
                        }
                    }
                }
                .padding(16)
            }
            .background(Color.appSurface)
            .navigationTitle("HR Department")
            .navigationBarTitleDisplayMode(.inline)
            .toolbar {
                ToolbarItem(placement: .topBarTrailing) {
                    Button("Done") { onDismiss() }
                }
            }
        }
    }

    private var headerBlock: some View {
        VStack(alignment: .leading, spacing: 4) {
            HStack(spacing: 6) {
                Text(">_")
                    .font(.system(size: 16, weight: .heavy, design: .monospaced))
                    .foregroundStyle(Color(hex: 0x00E676))
                Text("HR DEPARTMENT")
                    .font(.system(size: 18, weight: .black, design: .monospaced))
                    .tracking(2)
                    .foregroundStyle(Color.appTextPrimary)
            }
            Text("AGENT PERFORMANCE REVIEW")
                .font(.system(size: 10, weight: .heavy, design: .monospaced))
                .tracking(1.5)
                .foregroundStyle(Color.appTextSecondary)
                .padding(.leading, 22)
        }
    }

    private var losersSummary: String {
        let saved = Int(abs(losersTotal).rounded())
        let after = Int((totalBankroll - losersTotal).rounded())
        return "\(losers.count) agent\(losers.count == 1 ? " is" : "s are") costing money. Firing \(losers.count == 1 ? "them" : "all") saves $\(saved) and brings bankroll to $\(after)."
    }

    private var emptyState: some View {
        Text("No agents with settled picks yet. Check back after picks are graded.")
            .font(.system(size: 13, design: .monospaced))
            .foregroundStyle(Color.appTextSecondary)
            .multilineTextAlignment(.center)
            .frame(maxWidth: .infinity)
            .padding(.vertical, 32)
    }

    private func summaryCard(title: String, tint: Color, total: Double, body: String) -> some View {
        VStack(alignment: .leading, spacing: 6) {
            HStack {
                Text(title)
                    .font(.system(size: 10, weight: .heavy, design: .monospaced))
                    .tracking(1.5)
                    .foregroundStyle(tint)
                Spacer()
                Text(Self.formatDollars(total))
                    .font(.system(size: 16, weight: .black, design: .monospaced))
                    .foregroundStyle(tint)
            }
            Text(body)
                .font(.system(size: 13, design: .monospaced))
                .foregroundStyle(Color.appTextSecondary)
        }
        .padding(14)
        .background(
            RoundedRectangle(cornerRadius: 12, style: .continuous)
                .fill(tint.opacity(0.07))
        )
        .overlay(
            RoundedRectangle(cornerRadius: 12, style: .continuous)
                .strokeBorder(tint.opacity(0.18), lineWidth: 1)
        )
    }

    private func reportRow(_ card: ReportCard) -> some View {
        VStack(alignment: .leading, spacing: 8) {
            HStack(spacing: 10) {
                ZStack {
                    RoundedRectangle(cornerRadius: 8, style: .continuous)
                        .fill(card.grade.color.opacity(0.18))
                        .overlay(
                            RoundedRectangle(cornerRadius: 8, style: .continuous)
                                .strokeBorder(card.grade.color.opacity(0.4), lineWidth: 1)
                        )
                    Text(card.grade.rawValue)
                        .font(.system(size: 16, weight: .black, design: .monospaced))
                        .foregroundStyle(card.grade.color)
                }
                .frame(width: 32, height: 32)

                Text(card.agent.agent.avatarEmoji).font(.system(size: 20))

                VStack(alignment: .leading, spacing: 1) {
                    Text(card.agent.agent.name)
                        .font(.system(size: 14, weight: .bold))
                        .foregroundStyle(Color.appTextPrimary)
                        .lineLimit(1)
                    Text("\(card.record) | \(card.winRate.map { String(format: "%.1f%%", $0) } ?? "--")")
                        .font(.system(size: 11, weight: .semibold, design: .monospaced))
                        .foregroundStyle(Color.appTextSecondary)
                }

                Spacer(minLength: 4)

                VStack(alignment: .trailing, spacing: 1) {
                    Text(Self.formatDollars(card.dollarImpact))
                        .font(.system(size: 14, weight: .black, design: .monospaced))
                        .foregroundStyle(card.netUnits >= 0 ? Color(hex: 0x00E676) : Color(hex: 0xFF5252))
                    Text("IMPACT")
                        .font(.system(size: 7, weight: .heavy, design: .monospaced))
                        .tracking(1)
                        .foregroundStyle(Color.appTextSecondary)
                }
            }

            Text(card.recommendation)
                .font(.system(size: 12, design: .monospaced))
                .foregroundStyle(Color.appTextSecondary)

            if card.isCostingMoney {
                let companyWithoutFmt = Int(card.companyWithout.rounded())
                let savingsFmt = Int(abs(card.dollarImpact).rounded())
                HStack(alignment: .top, spacing: 6) {
                    Text(">_")
                        .font(.system(size: 10, weight: .heavy, design: .monospaced))
                        .foregroundStyle(Color(hex: 0xFF5252))
                    Text("Without \(card.agent.agent.name), bankroll would be ")
                        .font(.system(size: 11, design: .monospaced))
                        .foregroundStyle(Color(hex: 0xFF5252))
                    + Text("$\(companyWithoutFmt)")
                        .font(.system(size: 11, weight: .black, design: .monospaced))
                        .foregroundStyle(Color(hex: 0x00E676))
                    + Text(" (+$\(savingsFmt) saved)")
                        .font(.system(size: 11, design: .monospaced))
                        .foregroundStyle(Color(hex: 0xFF5252))
                }
                .padding(10)
                .background(
                    RoundedRectangle(cornerRadius: 8, style: .continuous)
                        .fill(Color(hex: 0xFF5252).opacity(0.08))
                )
                .overlay(
                    RoundedRectangle(cornerRadius: 8, style: .continuous)
                        .strokeBorder(Color(hex: 0xFF5252).opacity(0.15), lineWidth: 1)
                )
            }
        }
        .padding(12)
        .background(
            RoundedRectangle(cornerRadius: 12, style: .continuous)
                .fill(Color.appSurfaceElevated)
        )
        .overlay(
            RoundedRectangle(cornerRadius: 12, style: .continuous)
                .strokeBorder(Color.appBorder.opacity(0.4), lineWidth: 1)
        )
    }

    // MARK: - Helpers

    private static func grade(forNetUnits n: Double) -> Grade {
        if n >= 10 { return .s }
        if n >= 5 { return .a }
        if n >= 1 { return .b }
        if n >= 0 { return .c }
        if n >= -3 { return .d }
        return .f
    }

    private static func recommendation(grade: Grade, name: String, netUnits: Double, winRate: Double?) -> String {
        switch grade {
        case .s: return "\(name) is your MVP. Protect at all costs."
        case .a: return "\(name) is a top performer. Keep them running."
        case .b: return "\(name) is solid. Pulling their weight."
        case .c:
            if let wr = winRate, wr >= 50 {
                return "\(name) is break-even. Could tweak personality params."
            }
            return "\(name) is on thin ice. Review their strategy."
        case .d: return "\(name) is underperforming. Consider adjusting or replacing."
        case .f:
            let dollars = Int(abs(netUnits * unitSize).rounded())
            return "\(name) is costing you money. Fire to save $\(dollars)."
        }
    }

    private static func formatDollars(_ amount: Double) -> String {
        let abs = Swift.abs(amount)
        let sign = amount >= 0 ? "+" : "-"
        if abs >= 1000 {
            let k = abs / 1000.0
            return k.truncatingRemainder(dividingBy: 1) == 0
                ? "\(sign)$\(Int(k))k"
                : String(format: "%@$%.1fk", sign, k)
        }
        return String(format: "%@$%.0f", sign, abs)
    }
}

File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/agents/sheets/AgentHRBottomSheet.kt
package com.wagerproof.app.features.agents.sheets

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.ModalBottomSheet
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.material3.rememberModalBottomSheetState
import androidx.compose.runtime.Composable
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.SpanStyle
import androidx.compose.ui.text.buildAnnotatedString
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.withStyle
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.wagerproof.core.design.tokens.AppColors
import com.wagerproof.core.models.AgentWithPerformance
import java.util.Locale
import kotlin.math.abs
import kotlin.math.roundToInt

/**
 * Native port of iOS `AgentHRBottomSheet.swift` (itself a port of the web
 * `AgentHRBottomSheet.tsx`).
 *
 * "HR DEPARTMENT" comparison sheet: ranks every agent that has settled picks
 * by net units, grades them, and surfaces a recommendation. Includes a
 * "fire to save $$$" panel for losers. Pure visualization — no mutations.
 */

private const val UNIT_SIZE: Double = 100.0

private enum class HRGrade(val label: String, val color: Color) {
    S("S", Color(0xFFFFD700)),
    A("A", Color(0xFF00E676)),
    B("B", Color(0xFF69F0AE)),
    C("C", Color(0xFFFFC107)),
    D("D", Color(0xFFFF9800)),
    F("F", Color(0xFFFF5252)),
}

/** Report-card row derived per agent — mirrors the swift private struct. */
private data class HRReportCard(
    val agent: AgentWithPerformance,
    val grade: HRGrade,
    val netUnits: Double,
    val dollarImpact: Double,
    val winRate: Double?,
    val record: String,
    val companyWithout: Double,
    val recommendation: String,
    val isCostingMoney: Boolean,
) {
    val id: String get() = agent.id
}

private fun gradeForNetUnits(n: Double): HRGrade = when {
    n >= 10 -> HRGrade.S
    n >= 5 -> HRGrade.A
    n >= 1 -> HRGrade.B
    n >= 0 -> HRGrade.C
    n >= -3 -> HRGrade.D
    else -> HRGrade.F
}

private fun recommendation(grade: HRGrade, name: String, netUnits: Double, winRate: Double?): String =
    when (grade) {
        HRGrade.S -> "$name is your MVP. Protect at all costs."
        HRGrade.A -> "$name is a top performer. Keep them running."
        HRGrade.B -> "$name is solid. Pulling their weight."
        HRGrade.C ->
            if (winRate != null && winRate >= 50) "$name is break-even. Could tweak personality params."
            else "$name is on thin ice. Review their strategy."
        HRGrade.D -> "$name is underperforming. Consider adjusting or replacing."
        HRGrade.F -> {
            val dollars = abs(netUnits * UNIT_SIZE).roundToInt()
            "$name is costing you money. Fire to save \$$dollars."
        }
    }

/** Signed dollar formatter — "+\$1.2k" / "-\$450" mirroring the swift helper. */
private fun formatDollars(amount: Double): String {
    val a = abs(amount)
    val sign = if (amount >= 0) "+" else "-"
    if (a >= 1000) {
        val k = a / 1000.0
        return if (k % 1.0 == 0.0) "$sign\$${k.toInt()}k"
        else String.format(Locale.US, "%s\$%.1fk", sign, k)
    }
    return String.format(Locale.US, "%s\$%.0f", sign, a)
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun AgentHRBottomSheet(
    agents: List<AgentWithPerformance>,
    onDismiss: () -> Unit,
) {
    val sheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true)

    // Derive all report cards + section totals once per roster snapshot. This
    // is pure math over the cached performance rows — no fetching.
    val cards = remember(agents) {
        val totalNetUnits = agents.sumOf { it.performance?.netUnits ?: 0.0 }
        val totalBankroll = 1000 + totalNetUnits * UNIT_SIZE
        agents
            .filter { (it.performance?.totalPicks ?: 0) > 0 }
            .map { agent ->
                val perf = agent.performance!!
                val settled = perf.wins + perf.losses
                val winRate = if (settled > 0) perf.wins.toDouble() / settled * 100 else null
                val grade = gradeForNetUnits(perf.netUnits)
                HRReportCard(
                    agent = agent,
                    grade = grade,
                    netUnits = perf.netUnits,
                    dollarImpact = perf.netUnits * UNIT_SIZE,
                    winRate = winRate,
                    record = perf.recordLabel,
                    companyWithout = totalBankroll - perf.netUnits * UNIT_SIZE,
                    recommendation = recommendation(grade, agent.agent.name, perf.netUnits, winRate),
                    isCostingMoney = perf.netUnits < 0,
                )
            }
            .sortedBy { it.netUnits }
    }

    val winners = remember(cards) { cards.filter { !it.isCostingMoney }.sortedByDescending { it.netUnits } }
    val losers = remember(cards) { cards.filter { it.isCostingMoney } }
    val winnersTotal = remember(winners) { winners.sumOf { it.dollarImpact } }
    val losersTotal = remember(losers) { losers.sumOf { it.dollarImpact } }
    val totalBankroll = remember(agents) {
        1000 + agents.sumOf { it.performance?.netUnits ?: 0.0 } * UNIT_SIZE
    }

    ModalBottomSheet(
        onDismissRequest = onDismiss,
        sheetState = sheetState,
        containerColor = AppColors.appSurface,
    ) {
        // Title bar mirroring the swift NavigationStack inline title + Done.
        Row(
            modifier = Modifier.fillMaxWidth().padding(horizontal = 16.dp, vertical = 4.dp),
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Text(
                "HR Department",
                color = AppColors.appTextPrimary,
                fontSize = 17.sp,
                fontWeight = FontWeight.SemiBold,
            )
            Spacer(Modifier.weight(1f))
            TextButton(onClick = onDismiss) {
                Text("Done", color = AppColors.appPrimary, fontWeight = FontWeight.SemiBold)
            }
        }

        Column(
            modifier = Modifier
                .fillMaxWidth()
                .verticalScroll(rememberScrollState())
                .padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(16.dp),
        ) {
            HeaderBlock()

            if (winners.isEmpty() && losers.isEmpty()) {
                EmptyState()
            }

            if (winners.isNotEmpty()) {
                SummaryCard(
                    title = "WINNERS",
                    tint = Color(0xFF00E676),
                    total = winnersTotal,
                    body = "${winners.size} agent${if (winners.size == 1) "" else "s"} earning money for the company.",
                )
                winners.forEach { ReportRow(it) }
            }

            if (losers.isNotEmpty()) {
                val saved = abs(losersTotal).roundToInt()
                val after = (totalBankroll - losersTotal).roundToInt()
                val body = "${losers.size} agent${if (losers.size == 1) " is" else "s are"} costing money. " +
                    "Firing ${if (losers.size == 1) "them" else "all"} saves \$$saved and brings bankroll to \$$after."
                SummaryCard(title = "LOSERS", tint = Color(0xFFFF5252), total = losersTotal, body = body)
                losers.forEach { ReportRow(it) }
            }

            Spacer(Modifier.height(8.dp))
        }
    }
}

@Composable
private fun HeaderBlock() {
    Column(verticalArrangement = Arrangement.spacedBy(4.dp)) {
        Row(
            horizontalArrangement = Arrangement.spacedBy(6.dp),
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Text(
                ">_",
                color = Color(0xFF00E676),
                fontSize = 16.sp,
                fontWeight = FontWeight.Black,
                fontFamily = FontFamily.Monospace,
            )
            Text(
                "HR DEPARTMENT",
                color = AppColors.appTextPrimary,
                fontSize = 18.sp,
                fontWeight = FontWeight.Black,
                fontFamily = FontFamily.Monospace,
                letterSpacing = 2.sp,
            )
        }
        Text(
            "AGENT PERFORMANCE REVIEW",
            color = AppColors.appTextSecondary,
            fontSize = 10.sp,
            fontWeight = FontWeight.Black,
            fontFamily = FontFamily.Monospace,
            letterSpacing = 1.5.sp,
            modifier = Modifier.padding(start = 22.dp),
        )
    }
}

@Composable
private fun EmptyState() {
    Text(
        "No agents with settled picks yet. Check back after picks are graded.",
        color = AppColors.appTextSecondary,
        fontSize = 13.sp,
        fontFamily = FontFamily.Monospace,
        textAlign = TextAlign.Center,
        modifier = Modifier.fillMaxWidth().padding(vertical = 32.dp),
    )
}

@Composable
private fun SummaryCard(title: String, tint: Color, total: Double, body: String) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .background(tint.copy(alpha = 0.07f), RoundedCornerShape(12.dp))
            .border(1.dp, tint.copy(alpha = 0.18f), RoundedCornerShape(12.dp))
            .padding(14.dp),
        verticalArrangement = Arrangement.spacedBy(6.dp),
    ) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Text(
                title,
                color = tint,
                fontSize = 10.sp,
                fontWeight = FontWeight.Black,
                fontFamily = FontFamily.Monospace,
                letterSpacing = 1.5.sp,
            )
            Spacer(Modifier.weight(1f))
            Text(
                formatDollars(total),
                color = tint,
                fontSize = 16.sp,
                fontWeight = FontWeight.Black,
                fontFamily = FontFamily.Monospace,
            )
        }
        Text(body, color = AppColors.appTextSecondary, fontSize = 13.sp, fontFamily = FontFamily.Monospace)
    }
}

@Composable
private fun ReportRow(card: HRReportCard) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .background(AppColors.appSurfaceElevated, RoundedCornerShape(12.dp))
            .border(1.dp, AppColors.appBorder.copy(alpha = 0.4f), RoundedCornerShape(12.dp))
            .padding(12.dp),
        verticalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        Row(
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(10.dp),
        ) {
            // Grade chip
            Box(
                modifier = Modifier
                    .size(32.dp)
                    .background(card.grade.color.copy(alpha = 0.18f), RoundedCornerShape(8.dp))
                    .border(1.dp, card.grade.color.copy(alpha = 0.4f), RoundedCornerShape(8.dp)),
                contentAlignment = Alignment.Center,
            ) {
                Text(
                    card.grade.label,
                    color = card.grade.color,
                    fontSize = 16.sp,
                    fontWeight = FontWeight.Black,
                    fontFamily = FontFamily.Monospace,
                )
            }

            Text(card.agent.agent.avatarEmoji, fontSize = 20.sp)

            Column(modifier = Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(1.dp)) {
                Text(
                    card.agent.agent.name,
                    color = AppColors.appTextPrimary,
                    fontSize = 14.sp,
                    fontWeight = FontWeight.Bold,
                    maxLines = 1,
                )
                val wrLabel = card.winRate?.let { String.format(Locale.US, "%.1f%%", it) } ?: "--"
                Text(
                    "${card.record} | $wrLabel",
                    color = AppColors.appTextSecondary,
                    fontSize = 11.sp,
                    fontWeight = FontWeight.SemiBold,
                    fontFamily = FontFamily.Monospace,
                )
            }

            Column(horizontalAlignment = Alignment.End, verticalArrangement = Arrangement.spacedBy(1.dp)) {
                Text(
                    formatDollars(card.dollarImpact),
                    color = if (card.netUnits >= 0) Color(0xFF00E676) else Color(0xFFFF5252),
                    fontSize = 14.sp,
                    fontWeight = FontWeight.Black,
                    fontFamily = FontFamily.Monospace,
                )
                Text(
                    "IMPACT",
                    color = AppColors.appTextSecondary,
                    fontSize = 7.sp,
                    fontWeight = FontWeight.Black,
                    fontFamily = FontFamily.Monospace,
                    letterSpacing = 1.sp,
                )
            }
        }

        Text(
            card.recommendation,
            color = AppColors.appTextSecondary,
            fontSize = 12.sp,
            fontFamily = FontFamily.Monospace,
        )

        if (card.isCostingMoney) {
            val companyWithoutFmt = card.companyWithout.roundToInt()
            val savingsFmt = abs(card.dollarImpact).roundToInt()
            val red = Color(0xFFFF5252)
            val green = Color(0xFF00E676)
            // Multi-color inline sentence — "without X, bankroll would be $Y (+$Z saved)".
            val text = buildAnnotatedString {
                withStyle(SpanStyle(color = red, fontWeight = FontWeight.Black)) { append(">_ ") }
                withStyle(SpanStyle(color = red)) { append("Without ${card.agent.agent.name}, bankroll would be ") }
                withStyle(SpanStyle(color = green, fontWeight = FontWeight.Black)) { append("\$$companyWithoutFmt") }
                withStyle(SpanStyle(color = red)) { append(" (+\$$savingsFmt saved)") }
            }
            Text(
                text,
                fontSize = 11.sp,
                fontFamily = FontFamily.Monospace,
                modifier = Modifier
                    .fillMaxWidth()
                    .background(red.copy(alpha = 0.08f), RoundedCornerShape(8.dp))
                    .border(1.dp, red.copy(alpha = 0.15f), RoundedCornerShape(8.dp))
                    .padding(10.dp),
            )
        }
    }
}
